src/stdlib/contains_all.rs

```diff
@@ -0,0 +1,161 @@
+use crate::compiler::prelude::*;
+use crate::stdlib::string_utils::convert_to_string;
+
+fn contains_all(value: Value, substrings: Value, case_sensitive: Option<Value>) -> Resolved {
+    let to_lowercase = match case_sensitive {
+        Some(v) => v.try_boolean()?,
+        None => true,
+    };
+
+    let value_string = convert_to_string(value, to_lowercase)?;
+    let substring_values = substrings.try_array()?;
+
+    for substring_value in substring_values {
+        let substring = convert_to_string(substring_value, to_lowercase)?;
+        if !value_string.contains(&substring) {
+            return Ok(false.into());
+        }
+    }
+    Ok(true.into())
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ContainsAll;
+
+impl Function for ContainsAll {
+    fn identifier(&self) -> &'static str {
+        "contains_all"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "substrings",
+                kind: kind::ARRAY,
```

Comment:
I almost missed this one. The `substrings` input is only guaranteed to be an array when called, not necessarily an array _of strings_. As written, this function should be fallible since the `convert_to_string` could fail. I'd suggest adding a check in `compile` function.

Reply:
See this related issue: https://github.com/vectordotdev/vrl/issues/261
I keep seeing this come up. We should probably find a better way to prevent this since it's easy to miss. Fuzz testing can catch this, but maybe the `kind::ARRAY` should be changed to allow specifying the element type, or something like that.

Reply:
Updated the code, I like the idea of specifying the ARRAY type. I will make an issue.